packages/components/src/Typeahead/Typeahead.component.renderers.js

```diff
@@ -27,7 +30,7 @@ export function renderInputComponent(props) {
 	} = props;
 
 	const hasCaret = caret && !disabled && !readOnly;
-	const hasIcon = icon || hasCaret;
+	const hasIcon = icon;
```

Comment:
You can remove this const, just use `icon`

Reply:
re-oops